src/app/clusters/thermostat-server/thermostat-delegate.h

```diff
@@ -64,6 +64,12 @@ class Delegate
      */
     virtual CHIP_ERROR GetPresetAtIndex(size_t index, PresetStructWithOwnedMembers & preset) = 0;
 
+    /**
+     * @brief Creates a new pending presets list.
```

Comment:
I think this needs a lot more documentation.  For example, are callers allowed to call GetPendingPresetAtIndex before calling this function?  What about AppendToPendingPresetList?

In either case, this function should be down near the other pending preset functions.

And it should make it clear that it sets pending presets to current presets, at least in the documentation if not the naming.